### PR TITLE
A few argument related changes

### PR DESCRIPTION
**Pull request by ahal**
This series is definitely more on the optional side. I'm sort of fixing things as I go along and then spinning changes off into feature branches to get them landed earlier and avoid bitrot. Let me know if these are ok!

In case you were wondering, I have my main 'use_mozrunner' branch based on top of these and I'm getting close to being able to push it up for review.

### CODE REVIEW

File: raptor/raptor.py [hunk 1]
```diff
@@ -25,42 +25,36 @@
...
+        install_webext(self.app, self.profile)
 
     def start_control_server(self):
         self.control_server = RaptorControlServer()
         self.control_server.start()
 
-    def run_test(self):
-        start_browser(self.browser, self.browser_path, self.profile)
+    def run_test(self, test):
+        gen_test_url(self.app, test, self.sysdir)
```
Comment (rwood-moz) [inline raptor/raptor.py]: I believe this won't work - gen_test_url has to be done before the WebExt is installed - as gen_test_url writes a background script into the web extension folder, that is then loaded when the web ext is installed.

There may have already been a /webext/raptor-firefox/augo_gen_settings_url.js file left over from before and that's why this worked.

The reason I write this file into the web extension, is that's the only way to get the test name into the extension (then it queries the control server for the info for that test name).

The other alternative is I pass the test name into the control server init - however I didn't do it that way as I thought it best to keep the control server instance independent of the test being run. 

However we are also creating the profile now at the time of Raptor init, so maybe it wouldn't hurt to tie in the test name to the control server init also in that we probably will just be using the same Raptor and control server instance once anyway.

Comment (ahal) [reply raptor/raptor.py]: Ah, yeah sorry I didn't test this properly. Could we just also call `install_webext` implicitly in `run_test`? I'll re-push shortly with an example of what I mean.

As an aside, I think being able to run multiple tests from the same Raptor instance would prove useful in the future. Imagine this framework running tens of 'micro benchmarks' all in the same job. Having to re-create everything each time would be a waste. I think if Raptor is a success, it means adding new perf tests will be easy for developers to do, and in that world one test per job doesn't scale.

Comment (rwood-moz) [reply raptor/raptor.py]: Yes that works fine and actually I like that better, no need to split out each step into it's own method.

Good insight thanks and I agree. I wasn't sure if we would require a new totally clean browser profile before each test though (and we're currently creating the profile on Raptor init - however wouldn't be hard to just call create_profile again anyway).

**Review by rwood-moz — APPROVED** [review]
Thanks Andrew, appreciated!

**Pull request merged**